tests/unit_tests/audit_log/test_with_python_runner.py

```diff
@@ -0,0 +1,109 @@
+import contextlib
+from typing import (
+    Callable,
+    Union,
+)
+from unittest.mock import Mock
+
+import pytest
+
+from exasol.analytics.audit.audit_query_handler import (
+    AuditQueryHandler,
+    ParameterType,
+    ResultType,
+)
+from exasol.analytics.query_handler.context.scope import ScopeQueryHandlerContext
+from exasol.analytics.query_handler.context.top_level_query_handler_context import (
+    TopLevelQueryHandlerContext,
+)
+from exasol.analytics.query_handler.python_query_handler_runner import (
+    PythonQueryHandlerRunner,
+)
+from exasol.analytics.query_handler.query.result.interface import QueryResult
+from exasol.analytics.query_handler.query_handler import QueryHandler
+from exasol.analytics.query_handler.result import (
+    Continue,
+    Finish,
+)
+from exasol.analytics.schema import Column
+from exasol.analytics.sql_executor.testing.mock_sql_executor import MockSQLExecutor
+
+
+@pytest.fixture
+def context_mock(top_level_query_handler_context_mock) -> TopLevelQueryHandlerContext:
+    return top_level_query_handler_context_mock
+
+
+class SampleInput:
+    pass
+
+
+class SampleOutput:
+    def __init__(self, result: SampleInput):
+        self.result = result
+
+
+class StartFinishQh(QueryHandler[SampleInput, SampleOutput]):
+    def __init__(
+        self,
+        parameter: SampleInput,
+        context: ScopeQueryHandlerContext,
+    ):
+        super().__init__(parameter, context)
+        self._parameter = parameter
+
+    def start(self) -> Union[Continue, Finish[SampleOutput]]:
+        return Finish[SampleOutput](SampleOutput(self._parameter))
+
+    def handle_query_result(
+        self, query_result: QueryResult
+    ) -> Union[Continue, Finish[SampleOutput]]:
+        pass
+
+
+@contextlib.contextmanager
```

Comment:
why do you need a contextmanager, you are creating a function that doesn't use any resources